app/src/main/java/net/gsantner/markor/format/zimwiki/ZimWikiAutoFormat.java

```diff
@@ -0,0 +1,307 @@
+/*#######################################################
+ *
+ *   Maintained by Gregor Santner, 2018-
+ *   https://gsantner.net/
+ *
+ *   License of this file: Apache 2.0 (Commercial upon request)
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+#########################################################*/
+package net.gsantner.markor.format.zimwiki;
+
+import android.annotation.SuppressLint;
+import android.text.Editable;
+import android.text.InputFilter;
+import android.text.Spanned;
+
+import net.gsantner.opoc.util.StringUtils;
+
+import java.util.EmptyStackException;
+import java.util.Stack;
+import java.util.regex.Matcher;
+
+public class ZimWikiAutoFormat implements InputFilter {
+
+    @Override
+    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
+        try {
+            if (start < source.length() && dstart <= dest.length() && StringUtils.isNewLine(source, start, end)) {
+                return autoIndent(source, dest, dstart, dend);
+            }
+        } catch (IndexOutOfBoundsException | NullPointerException e) {
+            e.printStackTrace();
+        }
+        return source;
+    }
+
+    @SuppressLint("DefaultLocale")
+    private CharSequence autoIndent(final CharSequence source, final Spanned dest, final int dstart, final int dend) {
+
+        final String checkSymbol = "[ ] ";
+
+        final OrderedListLine oLine = new OrderedListLine(dest, dstart);
+        final UnOrderedListLine uLine = new UnOrderedListLine(dest, dstart);
+        final String indent = source + StringUtils.repeatChars(' ', oLine.indent);
+
+        final String result;
+        if (oLine.isOrderedList && oLine.lineEnd != oLine.groupEnd && dend >= oLine.groupEnd) {
+            result = indent + String.format("%d%c ", oLine.value + 1, oLine.delimiter);
+        } else if (uLine.isUnorderedList && uLine.lineEnd != uLine.groupEnd && dend >= uLine.groupEnd) {
+            final String checkString = uLine.isCheckboxList ? checkSymbol : "";
+            result = indent + String.format("%c %s", uLine.listChar, checkString);
+        } else {
+            result = indent;
+        }
+
+        return result;
+    }
+
+    public static class ListLine {
```

Comment:
We should try not to duplicate this code. If it is applicable to both zim as well as markdown, we should pull this code into a separate module and import it as appropriate.

Reply:
I think it just started off as copy-paste from markdown code

Reply:
I'll take a look. Thanks!

Reply:
closing this thread for now, we can take a look again when it's further progressed